src/ConfigCat.Cli/Commands/Scan.cs

```diff
@@ -153,7 +160,7 @@ internal class Scan(
                         PostLines = item.reference.PostLines,
                         PreLines = item.reference.PreLines,
                         ReferenceLine = item.reference.ReferenceLine
-                    }).ToList()
+                    }).Take(MaxReferenceCountToUpload).ToList()

```

Comment:
Maybe we could add a warning message if the references count is greater than MaxReferenceCountToUpload. Just to be sure.